tensorflow/c/experimental/filesystem/filesystem_interface.h

```diff
@@ -83,23 +83,25 @@ typedef struct TF_TransactionToken {
   TF_Filesystem* owner;
 } TF_TransactionToken;
 
-typedef struct TF_Filesystem_Option_Value {
-  int type_tag;
-  int num_values;
-  union {
-    int64_t inv_val;
-    double real_val;
-    struct {
-      char* buf;
-      int buf_length;
-    } buffer_val;
-  } * values;  // owned
+typedef union TF_Filesystem_Option_Value {
+  int64_t inv_val;
+  double real_val;
+  struct {
+    char* buf;
+    int buf_length;
```

Comment:
This should be a union to only contain one of the fields instead of all of them (since only one of them will be active, the others would be unitialized)

Reply:
Thanks @mihaimaruseac . The PR has been updated.